### ISSUE TITLE
Regalloc: MOV RSP

### ISSUE THREAD

**Issue by eponier** [jasmin-lang/jasmin#86]
In some cases (e.g. in tests/success/inline_regptr.jazz), regalloc leaves instructions such as `RAX = MOV(RSP)`. Would it be possible to optimize this case and replace `RAX` with `RSP` in the rest of the code (that does nothing fancy)? Is it even desirable?

**Comment by vbgl** [comment]
IIRC, register allocation never allocates anything to RSP.

**Comment by eponier** [comment]
Yes, the question is whether it should or not. I have no opinion on that.
